core/src/main/java/hivemall/topicmodel/OnlineLDAModel.java

```diff
@@ -0,0 +1,517 @@
+        double diff = 0.d;
+        for (int k = 0; k < _K; k++) {
+            diff += Math.abs(gammaPrev[k] - gammaNext[k]);
+        }
+        return (diff / _K) < _delta;
+    }
+
+    private void mStep() {
+        // calculate lambdaTilde for vocabularies in the current mini-batch
+        final Map<String, float[]> lambdaTilde = new HashMap<String, float[]>();
+        for (int d = 0; d < _miniBatchSize; d++) {
+            final Map<String, float[]> phi_d = _phi.get(d);
+            for (String label : _miniBatchMap.get(d).keySet()) {
+                float[] lambdaTilde_label;
+                if (lambdaTilde.containsKey(label)) {
+                    lambdaTilde_label = lambdaTilde.get(label);
+                } else {
+                    lambdaTilde_label = ArrayUtils.newInstance(_K, _eta);
+                }
+                lambdaTilde.put(label, lambdaTilde_label);
+
+                final float[] phi_label = phi_d.get(label);
+                for (int k = 0; k < _K; k++) {
+                    lambdaTilde_label[k] += _docRatio * phi_label[k];
+                }
+            }
+        }
+
+        // update lambda for all vocabularies
+        for (Map.Entry<String, float[]> e : _lambda.entrySet()) {
+            String label = e.getKey();
+            final float[] lambda_label = e.getValue();
+
+            float[] lambdaTilde_label;
+            if (lambdaTilde.containsKey(label)) {
+                lambdaTilde_label = lambdaTilde.get(label);
+            } else {
+                lambdaTilde_label = ArrayUtils.newInstance(_K, _eta);
+            }
+            _lambda.put(label, lambda_label);
+
+            for (int k = 0; k < _K; k++) {
+                lambda_label[k] = (float) ((1.d - _rhot) * lambda_label[k] + _rhot
+                        * lambdaTilde_label[k]);
+            }
+        }
+    }
+
+    /**
+     * Calculate approximate perplexity for the current mini-batch.
+     */
+    public float computePerplexity() {
+        float bound = computeApproxBound();
+        float perWordBound = bound / (_docRatio * _wordCount);
+        return (float) Math.exp(-1.f * perWordBound);
+    }
+
+    /**
+     * Estimates the variational bound over all documents using only the documents passed as mini-batch.
+     */
+    private float computeApproxBound() {
+        float score = 0.f;
+
+        // prepare
+        final float[] gammaSum = new float[_miniBatchSize];
+        for (int d = 0; d < _miniBatchSize; d++) {
+            gammaSum[d] = MathUtils.sum(_gamma[d]);
+        }
+        final float[] lambdaSum = new float[_K];
+        for (float[] lambda : _lambda.values()) {
+            MathUtils.add(lambdaSum, lambda, _K);
+        }
+
+        for (int d = 0; d < _miniBatchSize; d++) {
+            // E[log p(doc | theta, beta)]
+            for (Map.Entry<String, Float> e : _miniBatchMap.get(d).entrySet()) {
+                final float[] lambda = _lambda.get(e.getKey());
```

Comment:
@takuti hmm.. 

For `Gamma.digamma(lambda[k]`, it's okey as it is.

Please rename variable name `lambda` to `lambdaD`.